### PR TITLE
[util] Decrease polling timeout accordingly after long-running steps

### PR DESCRIPTION
**Pull request by valfirst**

### CODE REVIEW

File: vividus-util/src/test/java/org/vividus/util/wait/DurationBasedWaiterTests.java [hunk 1]
```diff
@@ -58,4 +58,19 @@ void shouldInvokeRunnableOnceBeforeReachingStopCondition() throws IOException
...
+    {
+        FailableSupplier<Boolean, IOException> valueProvider = mock(FailableSupplier.class);
+        when(valueProvider.get())
+                .thenAnswer(invocation -> {
+                    Thread.sleep(5_000);
+                    return false;
+                })
+                .thenReturn(true)
+                .thenReturn(false);
+        assertTrue(new DurationBasedWaiter(new WaitMode(Duration.ofSeconds(10), 10))
```
Comment (ikalinin1) [inline vividus-util/src/test/java/org/vividus/util/wait/DurationBasedWaiterTests.java:73]: does it make sense to validate the quantity of provider calls here?

Comment (valfirst) [reply vividus-util/src/test/java/org/vividus/util/wait/DurationBasedWaiterTests.java:75]: no, the result check guarantees the provider is invoked properly, other edge cases are covered by another unit tests

File: vividus-util/src/test/java/org/vividus/util/wait/DurationBasedWaiterTests.java [hunk 2]
```diff
@@ -58,4 +60,19 @@ void shouldInvokeRunnableOnceBeforeReachingStopCondition() throws IOException
...
         verify(failableRunnable, times(1)).run();
     }
+
+    @Test
+    void shouldDecreasePollingTimeoutAccordinglyAfterLongRunningRunnable() throws IOException
+    {
+        FailableSupplier<Boolean, IOException> valueProvider = mock(FailableSupplier.class);
+        when(valueProvider.get())
+                .thenAnswer(invocation -> {
+                    Sleeper.sleep(Duration.ofSeconds(5));
```
Comment (ikalinin1) [inline vividus-util/src/test/java/org/vividus/util/wait/DurationBasedWaiterTests.java:70]: Why is the timeout so big, any particular reason for this?

Comment (valfirst) [reply vividus-util/src/test/java/org/vividus/util/wait/DurationBasedWaiterTests.java:70]: runnable execution time is 5 times bigger than polling timeout, having such values we can guarantee no race condition will be met (especially on slow machines)

Comment (valfirst) [reply vividus-util/src/test/java/org/vividus/util/wait/DurationBasedWaiterTests.java:70]: another option is to mock `sleep`-s

Comment (valfirst) [reply vividus-util/src/test/java/org/vividus/util/wait/DurationBasedWaiterTests.java:70]: https://github.com/vividus-framework/vividus/search?q=slow&type=commits - in general to avoid such "fine"-tunnings

**Review by ikalinin1 — APPROVED** [review]

**Review by uarlouski — APPROVED** [review]
